src/fidesops/service/masking/strategy/masking_strategy_factory.py

```diff
@@ -10,70 +10,57 @@
-            cls[item]
-        except KeyError:
-            return False
+    def register(cls, name: str) -> Callable[[MaskingStrategy], MaskingStrategy]:
+        def wrapper(strategy_class: MaskingStrategy) -> MaskingStrategy:
+            if name in cls.registry:
+                logger.warning(
+                    f"Strategy with name {name} already exists in MaskingStrategy registry. It referred to class {cls.registry[name]}, but will now refer to {strategy_class}"
+                )
+            logger.info(
+                f"Registering MaskingStrategy class {strategy_class} under name {name}"
+            )
+            cls.registry[name] = strategy_class
+            cls.valid_strategies = ", ".join(cls.registry.keys())
+            return strategy_class
 
-        return True
-
-
-def get_strategy(
-    strategy_name: str,
-    configuration: Dict[
-        str,
-        Union[str, FormatPreservationConfig],
-    ],
-) -> MaskingStrategy:
-    """
-    Returns the strategy given the name and configuration.
-    Raises NoSuchStrategyException if the strategy does not exist
-    """
-    if not SupportedMaskingStrategies.__contains__(strategy_name):
-        valid_strategies = ", ".join([s.name for s in SupportedMaskingStrategies])
-        raise NoSuchStrategyException(
-            f"Strategy '{strategy_name}' does not exist. Valid strategies are [{valid_strategies}]"
-        )
-    strategy = SupportedMaskingStrategies[strategy_name].value
-    try:
-        strategy_config: MaskingConfiguration = strategy.get_configuration_model()(
-            **configuration
-        )
-        return strategy(configuration=strategy_config)
-    except ValidationError as e:
-        raise FidesopsValidationError(message=str(e))
+        return wrapper
 
+    @classmethod
+    def get_strategy(
+        cls,
+        strategy_name: str,
+        configuration: Dict[
+            str,
+            Union[str, FormatPreservationConfig],
+        ],
+    ) -> MaskingStrategy:
+        """
+        Returns the strategy given the name and configuration.
+        Raises NoSuchStrategyException if the strategy does not exist
+        """
+        if strategy_name not in cls.registry:
+            raise NoSuchStrategyException(
+                f"Strategy '{strategy_name}' does not exist. Valid strategies are [{cls.valid_strategies}]"
+            )
+        strategy: MaskingStrategy = cls.registry[strategy_name]
+        try:
+            strategy_config: MaskingConfiguration = strategy.get_configuration_model()(
+                **configuration
+            )
+            return strategy(configuration=strategy_config)
+        except ValidationError as e:
+            raise FidesopsValidationError(message=str(e))
 
-def get_strategies() -> List[MaskingStrategy]:
-    """Returns all supported masking strategies"""
-    return [e.value for e in SupportedMaskingStrategies]
+    @classmethod
+    def get_strategies(cls) -> ValuesView[MaskingStrategy]:
```

Comment:
This function gets called a lot, and is extremely simple. As a small optimization I think we can come up with a new class variable to house this list instead.

Reply:
Yea, more or less that.

Reply:
i can push something up like this (small updates to the factory registration)
```
    class MaskingStrategyFactory:
    registry: Dict[str, Type[MaskingStrategy]] = {}
    strategies: Set[Type[MaskingStrategy]] = set()
    valid_strategies_string: str = ""

    @classmethod
    def register(cls, name: str) -> Callable[[Type[MaskingStrategy]], Type[MaskingStrategy]]:
        def wrapper(strategy_class: Type[MaskingStrategy]) -> Type[MaskingStrategy]:
            logger.debug(
                f"Registering new masking strategy '{strategy_class}' under name '{name}'"
            )

            if name in cls.registry:
                logger.warning(
                    f"Masking strategy with name '{name}' already exists. It previously referred to class '{cls.registry[name]}', but will now refer to '{strategy_class}'"
                )
                cls.strategies.remove(cls.registry[name])

            cls.registry[name] = strategy_class
            cls.strategies.add(strategy_class)
            cls.valid_strategies_string = ", ".join(cls.registry.keys())
            return cls.registry[name]

        return wrapper
```
note the `strategies` set, and the calls to `remove` and `add` to it.

but what i don't love is that we're essentially maintaining a duplicate set of values in addition to what's backing the registry dict, and we have to "manually" keep them in sync. perhaps we could remove the method and just switch the callers to access the class's `registry.values()` directly. thoughts?

Reply:
Seeing it coded out and now I'm backtracking on my original comment 😞 I think it's fine to just leave the call to `.values()` in a class method, and prevent the extra bloat.